fileuploader/large_file_upload_task.go

```diff
@@ -0,0 +1,149 @@
+package fileuploader
+
+import (
+	"context"
+	"errors"
+	abstractions "github.com/microsoft/kiota-abstractions-go"
+	"github.com/microsoft/kiota-abstractions-go/serialization"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+type LargeFileUploadTask[T interface{}] interface {
```

Comment:
Since we know that T is a model generated by kiota, would it make sense to set the type as a Parsable?
```suggestion
type LargeFileUploadTask[T abstractions.Parsable] interface {
```